### PR TITLE
adding a few more test cases for entry parsing docker format

### PR DESCRIPTION
**Pull request by slim-bean**
just to augment the test cases that came in with the CRI format I added a few tests for the existing docker format

### CODE REVIEW

File: pkg/promtail/api/entry_parser_test.go [hunk 1]
```diff
@@ -18,58 +18,90 @@ type Entry struct {
...
-		EntryParser := CRI.Wrap(client)
-		err := EntryParser.Handle(model.LabelSet{}, time.Now(), tc.Line)
+		handler := parser.Wrap(client)
+		err := handler.Handle(model.LabelSet{}, time.Now(), tc.Line)
 		hasError := err != nil
 
 		if tc.Error != hasError {
-			t.Error("For", tc.Line, "expected", tc.Error, "got", hasError)
+			t.Fatal("Parse error for test case", i, "with entry", tc.Line, "\nerror:", err)
 		}
```
Comment (tomwilkie) [inline pkg/promtail/api/entry_parser_test.go]: Can you update the tests to use `require` and `assert` pls?

File: pkg/promtail/api/entry_parser_test.go [hunk 2]
```diff
@@ -18,59 +20,79 @@ type Entry struct {
...
-			if tc.Expected.Log != entry.Log {
-				t.Error("For", tc.Line, "expected", tc.Expected.Log, "got", entry.Log)
-			}
-
-			if !tc.Expected.Labels.Equal(entry.Labels) {
-				t.Error("For", tc.Line, "expected", tc.Expected.Labels, "got", entry.Labels)
-			}
+			assert.Equal(t, tc.Expected.Time, entry.Time, "Time error for test case %d, with entry %s", i, tc.Line)
+			assert.Equal(t, tc.Expected.Log, entry.Log, "Log entry error for test case %d, with entry %s", i, tc.Line)
+			assert.True(t, tc.Expected.Labels.Equal(entry.Labels), "Label error for test case %d, labels did not match; expected: %s, found %s", i, tc.Expected.Labels, entry.Labels)
```
Comment (tomwilkie) [inline pkg/promtail/api/entry_parser_test.go]: Why can't you do `assert.Equal(t, tc.Expected, entry)`?

File: pkg/promtail/api/entry_parser_test.go [hunk 3]
```diff
@@ -18,59 +20,79 @@ type Entry struct {
...
+		handler := parser.Wrap(client)
+		err := handler.Handle(model.LabelSet{}, time.Now(), tc.Line)
 		hasError := err != nil
 
 		if tc.Error != hasError {
-			t.Error("For", tc.Line, "expected", tc.Error, "got", hasError)
+			t.Fatal("Parse error for test case", i, "with entry", tc.Line, "\nerror:", err)
 		}
 
 		if !tc.Error {
```
Comment (tomwilkie) [inline pkg/promtail/api/entry_parser_test.go]: `else`?

File: pkg/promtail/api/entry_parser_test.go [hunk 4]
```diff
@@ -18,59 +20,79 @@ type Entry struct {
...
 			Entries: make([]Entry, 0),
 		}
 
-		EntryParser := CRI.Wrap(client)
-		err := EntryParser.Handle(model.LabelSet{}, time.Now(), tc.Line)
+		handler := parser.Wrap(client)
+		err := handler.Handle(model.LabelSet{}, time.Now(), tc.Line)
 		hasError := err != nil
 
 		if tc.Error != hasError {
```
Comment (tomwilkie) [inline pkg/promtail/api/entry_parser_test.go]: `if tc.Error == (err == nil) {`?

File: pkg/promtail/api/entry_parser_test.go (hunk 2 above)
Comment (slim-bean) [reply pkg/promtail/api/entry_parser_test.go]: Could simplify the first two, but I think I prefer the idea of using the prometheus Labels.Equal function for making sure the labels match?

File: pkg/promtail/api/entry_parser_test.go (hunk 4 above)
Comment (slim-bean) [reply pkg/promtail/api/entry_parser_test.go]: I went a slightly different route with this, let me know if it works ok for you

**Pull request merged**